cmd/admin-handlers.go

```diff
@@ -2575,6 +2579,135 @@ func fetchKMSStatus() madmin.KMS {
 	return kmsStat
 }
 
+// fetchKMSStatusV2 fetches KMS-related status information for all instances
+func fetchKMSStatusV2(ctx context.Context) []madmin.KMS {
+	if GlobalKMS == nil {
+		return []madmin.KMS{}
+	}
+	stat, err := GlobalKMS.Stat(ctx)
+	if err != nil {
+		return []madmin.KMS{}
+	}
+	if len(stat.Endpoints) == 0 {
+		return []madmin.KMS{}
+	}
+
+	kesClient := getKesClient()
+	stats := []madmin.KMS{}
+
+	kmsContext := kms.Context{"MinIO admin API": "ServerInfoHandler"} // Context for a test key operation
+	for _, endpoint := range stat.Endpoints {
+		client := kes.Client{
+			Endpoints:  []string{endpoint},
+			HTTPClient: kesClient.HTTPClient,
+		}
+		// 1. Get stats for the KES instance
+		state, err := client.Status(ctx)
+		if err != nil {
+			stats = append(stats, madmin.KMS{Status: string(madmin.ItemOffline), Endpoint: endpoint})
+			continue
+		}
+		stat := madmin.KMS{Status: string(madmin.ItemOnline), Endpoint: endpoint, Version: state.Version}
+
+		// 2. Generate a new key using the KMS.
+		kmsCtx, err := kmsContext.MarshalText()
+		if err != nil {
+			stats = append(stats, madmin.KMS{Status: string(madmin.ItemOffline), Endpoint: endpoint})
+			continue
+		}
+		key, err := client.GenerateKey(ctx, env.Get(kms.EnvKESKeyName, ""), kmsCtx)
+		if err != nil {
+			stat.Encrypt = fmt.Sprintf("Encryption failed: %v", err)
+		} else {
+			stat.Encrypt = "success"
+		}
+		// 3. Verify that we can indeed decrypt the (encrypted) key
+		decryptedKey, err := client.Decrypt(ctx, env.Get(kms.EnvKESKeyName, ""), key.Ciphertext, kmsCtx)
+		switch {
+		case err != nil:
+			stat.Decrypt = fmt.Sprintf("Decryption failed: %v", err)
+		case subtle.ConstantTimeCompare(key.Plaintext, decryptedKey) != 1:
+			stat.Decrypt = "Decryption failed: decrypted key does not match generated key"
+		default:
+			stat.Decrypt = "success"
+		}
+		stats = append(stats, stat)
+	}
+
+	return stats
+}
+
+// getKesClient returns an http client for KES
```

Comment:
this entire code is also repeated, not sure why we need this as well. We can add another API at 

```
GlobalKMS.Verify() ([]VerifyResult) 
```

```
type VerifyResult struct{
   Endpoint string
   Decrypt string
   Encrypt string
```

that does all of this and returns us the relevant information.

We can do this per endpoint as well.

Reply:
Done. Added a function `GlobalKMS.Verify` and used the same.